### ISSUE TITLE
[unbound-method] False positives when not invoking or assigning method

### ISSUE THREAD

**Issue by lostfictions** [typescript-eslint/typescript-eslint#1425]
`unbound-method` often (but not consistently) reports a false positive when simply checking for the existence of a method.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/unbound-method": "warn"
  }
}
```

Given this code:

```TS
interface OptionalMethod {
  mightBeDefined?(): void
}

const x: OptionalMethod = {};
const myCondition = true;
```

This produces a warning:

```ts
if(myCondition || x.mightBeDefined) {
  //              ^^^^^^^^^^^^^^^^
  //              Warning: Avoid referencing unbound methods...
  console.log('hello world')
}
```

However, this (correctly) does not show a warning:

```ts
if(x.mightBeDefined) {
  console.log('hello world')
}
```

Additionally, while technically a useless expression (and likely a code smell), simply referring to the method also displays a warning:
```ts
x.mightBeDefined;
^^^^^^^^^^^^^^^^
Warning: Avoid referencing unbound methods...
```

**Expected Result**

No warning when an unbound method is referred to in a context where it's not being assigned, invoked, or passed as an argument. 

**Actual Result**

A warning, in the scenarios outlined above.

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.15.0` |
| `@typescript-eslint/parser`        | `2.15.0` |
| `TypeScript`                       | `3.7.4` |
| `ESLint`                           | `6.8.0` |
| `node`                             | `13.1.0` |
| `npm`                              | `6.13.0` |
